torchmultimodal/modules/layers/transformer.py

```diff
@@ -0,0 +1,327 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+import math
+from collections import namedtuple
+from functools import partial
+from typing import Any, Callable, Optional
+
+import torch
+from torch import nn, Tensor
+from torchmultimodal.modules.layers.normalizations import Fp32LayerNorm
+
+TransformerOutput = namedtuple(
+    "TransformerOutput",
+    ["last_hidden_state", "pooler_output", "hidden_states", "attentions"],
+    defaults=(None, None, None, None),
+)
+
+
+class TransformerSelfAttention(nn.Module):
```

Comment:
This refactor will probably confused users more because as of now CLIP implementation is using Transformer from torchtext and then now we have these Transformer implementation. I would suggest possibly renaming them to `FlavaTransformer`, `FlavaSelfAttention` and so on or figure out the long term unification strategy.

Reply:
This is also one of the reasons why I kept it inside the flava model file itself to avoid contamination of lib's modules.

Reply:
I will rename to flava transformer for now until we figure out long term unification